api/app/controllers/spree/api/checkouts_controller.rb

```diff
@@ -51,11 +53,28 @@ def update
       end
 
       private
-
         def object_params
-          # For payment step, filter order parameters to produce the expected nested attributes for a single payment and its source, discarding attributes for payment methods other than the one selected
-          # respond_to check is necessary due to issue described in #2910
-          object_params = nested_params
+          modify_payment_attributes params[:order] || {}
+
+          protected_params = if params[:order]
+                               params.require(:order).permit(permitted_checkout_attributes)
+                             else
+                               {}
+                             end
+
+          map_nested_attributes_keys Order, protected_params
+        end
+
+        def user_id
+          params[:order][:user_id] if params[:order]
+        end
+
+        # For payment step, filter order parameters to produce the expected
+        # nested attributes for a single payment and its source, discarding
+        # attributes for payment methods other than the one selected
+        #
+        # respond_to check is necessary due to issue described in #2910
+        def modify_payment_attributes(object_params)
```

Comment:
Assignment Branch Condition size for modify_payment_attributes is too high. [22.83/15]<br>Inconsistent indentation detected.